src/core/qgslayerdefinition.cpp

```diff
@@ -51,6 +51,7 @@ bool QgsLayerDefinition::loadLayerDefinition( const QString &path, QgsProject *p
 
   QgsReadWriteContext context;
   context.setPathResolver( QgsPathResolver( path ) );
+  context.setProjectTranslator( project );
```

Comment:
I guess this it for opening a `.qlr` file. I'm not sure if we want to apply the translations from the currently loaded project or rather in the future have QGIS look for `.qm` translations next to the `.qlr` file itself... Opinions?

Reply:
I'm not sure. When I think about it, I guess I prefer not to translate it when loading the `.qlr`, because there is most possibly no translation available anyway (because the ts file has been created without it...). Or do you think someone does that on purpose?
I like your idea to have a `.qm` file for the `.qlr` itself...